### PR TITLE
extract 'unref_pubkeys' for shared code

### PR DESCRIPTION
**Pull request by jeffwashington**
#### Problem
share code across shrink, clean, and ancient append vecs

#### Summary of Changes

Fixes #

### CODE REVIEW

File: runtime/src/accounts_db.rs [hunk 1]
```diff
@@ -8078,6 +8075,23 @@ impl AccountsDb {
                 )
             });
         });
+    }
+
+    /// lookup each pubkey in 'purged_slot_pubkeys' and unref it in the accounts index
+    /// populate 'purged_stored_account_slots' by grouping 'purged_slot_pubkeys' by pubkey
+    // pubkeys_removed_from_accounts_index - These keys have already been removed from the accounts index
+    //    and should not be unref'd. If they exist in the accounts index, they are NEW.
```
Comment (apfitzge) [inline runtime/src/accounts_db.rs:8083]: are these 2 lines intentionally not doc comments?

File: runtime/src/accounts_db.rs [hunk 2]
```diff
@@ -4324,11 +4324,11 @@ impl AccountsDb {
             }
         });
 
-        self.thread_pool_clean.install(|| {
-            unref.into_par_iter().for_each(|key| {
-                self.accounts_index.unref_from_storage(key);
```
Comment (apfitzge) [inline runtime/src/accounts_db.rs:4329]: with this removal, `AccountsIndex::unref_from_storage` is unused.

**Review by apfitzge — APPROVED** [review]
Code changes look good - if you just update the comments to be /// no need to re-review.

File: runtime/src/accounts_db.rs (hunk 2 above)
Comment (jeffwashington) [reply runtime/src/accounts_db.rs:4329]: removed it

File: runtime/src/accounts_db.rs (hunk 1 above)
Comment (jeffwashington) [reply runtime/src/accounts_db.rs]: fixed all of them

**Review by apfitzge — APPROVED** [review]

**Pull request merged**